app/processing/index.js

```diff
@@ -1,20 +1,22 @@
 const { processingConfig } = require('../config')
-
+const waitForIdleMessaging = require('../messaging/wait-for-idle-messaging')
 const schedulePendingSettlements = require('./schedule')
-const { getStatement, sendStatement } = require('./statement')
+const { getStatement, sendStatement, validateStatement } = require('./statement')
+const updateScheduleByScheduleId = require('./statement/update-schedule-by-schedule-id')
 
 const start = async () => {
   try {
     if (processingConfig.constructionActive) {
       const pendingStatements = await schedulePendingSettlements()
-      if (!pendingStatements) {
-        throw new Error('No statements to be generated')
-      }
 
       for (const pendingStatement of pendingStatements) {
         try {
+          await waitForIdleMessaging()
           const aggregatedStatement = await getStatement(pendingStatement.settlementId)
-          await sendStatement(pendingStatement.scheduleId, aggregatedStatement)
+          if (validateStatement(aggregatedStatement)) {
+            await sendStatement(aggregatedStatement)
+          }
```

Comment:
If we have a recovery, ie no more money due to the customer then we will still receive the AP settlement for the original payment each quarter.  We'll just also get a counter settlement to cancel it out for the PPA.

We only want to send a statement if the customer is receiving a payment and as they're not in this scenario, we can safely just skip sending the statement.

Have moved the update schedule to here out of sending so that these schedules are still marked as completed to avoid an endless loop.